### PR TITLE
glusterfs: Glusterfs 3.12 and service cherry-picks

### PR DESCRIPTION
**Pull request by nh2**
###### Motivation for this change

Cherry-pick of #29868 and #29062 (as planned in those tickets).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

and

- [X] tested with https://github.com/nh2/nixops-gluster-example

---

CC @bachp

### CODE REVIEW

**Review by bachp — APPROVED** [review]

File: nixos/modules/services/network-filesystems/glusterfs.nix
Comment (bachp) [inline nixos/modules/services/network-filesystems/glusterfs.nix]: Must be `after = [ "network.target" "local-fs.target" ] ++ lib.optional cfg.useRpcbind "rpcbind.service";`

Comment (nh2) [reply nixos/modules/services/network-filesystems/glusterfs.nix]: Ah good catch, I thought I had fixed that already but the brackets are still there.

Thanks for fixing it!

**Comment by nh2** [comment]
I've cherry-picked @bachp's fix for the above mentioned issue.

This is ready to land now.

**Comment by nh2** [comment]
CC @joachifm

**Comment by joachifm** [comment]
Deferring to @fpletz

**Comment by FRidh** [comment]
In my opinion its a bit too late for a bump like this to make it into stable, but I suppose if the maintainer, @7c6f434c , is fine with this, then it should be ok?

**Comment by FRidh** [comment]
Note that the service does not have a maintainer. I think it should be added (on master, first).

**Comment by 7c6f434c** [comment]
The problem is that I don't use GlusterFS regularly nowadays, and I haven't used stable channels, hm, maybe not ever.

(Looking at the list of fixes… hm, a crash… a corruption if some combination of factors happens…) Maybe there are enough bugfixes for a bump, I dunno.

Looking at the file history, I would actually ask whether @nh2 wants to be added as the maintainer of both package and service.

**Comment by nh2** [comment]
Yes, I'd be happy to maintain gluster for a while.

I have a gluster deployment based on the current `release-*` stable channel for which I care that it doesn't break, and I maintain [nixops-gluster-example](https://github.com/nh2/nixops-gluster-example) with which changes to package + service, as well as version upgrades, can be publicly tested.

So far I know only of @bachp as another NixOS user of gluster, so I've CC'd him in the past so that he can review/test changes to be made. It would be nice to know if there are other users who depend on gluster so that those can be CC'd too.

**Comment by bachp** [comment]
I think my usage of gluster is a lot simpler than @nh2. I currently don't use any declarative configuration of gluster volumes but mainly depend on the service to have the gluster daemons running.
I think @nh2 you did a very good job so far improving gluster support in NixOS. I would also like to see a test for glusterfs that is based on the nixops setup mentioned above.

**Comment by nh2** [comment]
Uh I have some confusion here.

The PR is still open but I noticed that its commits have landed on `release-17.09` by now, for example https://github.com/NixOS/nixpkgs/commit/1be7119b20370017469d6f6988a597da5fb0f142

I noticed this only by the Github info message [here](https://github.com/NixOS/nixpkgs/pull/29062#ref-commit-1be7119):

> fpletz added a commit that referenced this pull request 8 days ago

I'm of course OK with those picks since I requested them here, but I do have two questions about it:

* The first one probably to @fpletz: How comes the individual commits are interleaved with other commits, as opposed to be one after the other like in the PR? Is there some tooling involved that I don't know about yet?
* How can I learn that such picks happened, so that I can do cleanup actions like close this PR?

---

(I also noticed that I typoed the PR title to glusterfs `1.2` instead of `3.12`, will fix that now.)

**Comment by fpletz** [comment]
@nh2 As release manager, I regularly go through the changes in master and pick relevant fixes to the release branch. Unfortunately, I was a bit busy the last few weeks so I wasn't also monitoring incoming pull requests. Sorry about that.

There isn't any kind of notification from github when commits are picked. However, if a PR or issue is mentioned in the commit message, cherry-picking the commit triggers a message. No notification via email, though.

After trying to merge this, there is no diff to release-17.09, so we can safely close this. Thanks a lot for your efforts!

**Comment by nh2** [comment]
@fpletz OK, thank you!

**Pull request closed without merging**
